### ISSUE TITLE
BadBufferTest.test_truncated_shm_file fails intermittently with `unaligned tcache chunk detected`

### ISSUE THREAD

**Issue by RAOF** [MirServer/mir#1975]
If you run `BadBufferTest.test_truncated_shm_file` in a loop (*not* using `gtest_repeat`, as Wayland's SIGSEGV handler is only initialised the first time in a process) on 21.04 then it will *eventually* fail with:
```
Note: Google Test filter = BadBufferTest.test_truncated_shm_file
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from BadBufferTest
[ RUN      ] BadBufferTest.test_truncated_shm_file
error in client communication (pid 1539687)
wl_buffer@15: error 2: error accessing SHM buffer
tcache_thread_shutdown(): unaligned tcache chunk detected
fish: “/usr/lib/x86_64-linux-gnu/wlcs/…” terminated by signal SIGABRT (Abort)
```

This can take a while; I've seen it happen as soon as 30 iterations, or after > 3,000 iterations.

**Comment by Saviq** [comment]
Confirmed through:
```shell
I=0; while /usr/libexec/wlcs/wlcs ./build-Debug/lib/miral_wlcs_integration.so --gtest_filter=BadBufferTest.test_truncated_shm_file; do I=$(( $I + 1 )); done;
```

It failed after some 20k iterations.

**Comment by Saviq** [comment]
I've been 60k iterations in now, half of it under `stress`. Calling this fixed by #2744.
